### PR TITLE
Update ApplicationCache for v8

### PR DESCRIPTION
**Pull request by erikjanwestendorp**
Belongs to: https://github.com/umbraco/UmbracoDocs/issues/1874

### CODE REVIEW

File: Reference/Cache/applicationcache.md [hunk 1]
```diff
@@ -1,31 +1,57 @@
...
 
 ## Cache types
 
-The `Umbraco.Core.CacheHelper` which can be found in namespace `Umbraco.Core.Cache` contains 3 types of cache: Runtime Cache, Request Cache and Static Cache.
+The `AppCaches` which can be found in namespace `Umbraco.Core.Cache` contains servaral types of cache: Runtime Cache, Request Cache and Isolated Cache.
 
 **Runtime Cache** is the most commonly used and is synonymous with HttpRuntime.Cache.
-**Request cache** is cache that exists only for the current request, this is synonymous with HttpContext.Current.Items and **static cache** is cache that exists globally and never expires. Static cache should be rarely used and should be used with caution since it can cause memory issues.
+**Request cache** is cache that exists only for the current request, this is synonymous with HttpContext.Current.Items and **isolated caches** are used by e.g. repositories, to ensure that each cached entity type has its own cache, so that lookups are fast and the repository does not need to search through all keys on a global scale.
```
Comment (sofietoft) [inline Reference/Cache/applicationcache.md:14]:
```suggestion
**Request cache** is cache that exists only for the current request. This is synonymous with HttpContext.Current.Items and **isolated caches** are used by e.g. repositories, to ensure that each cached entity type has its own cache. When they have their own cache lookups are fast and the repository does not need to search through all keys on a global scale.
```

File: Reference/Cache/applicationcache.md [hunk 2]
```diff
@@ -1,31 +1,57 @@
...
-The documentation available around caching in the current Umbraco version can be found here: [Caching](../Cache).
-:::
-
-You should always be doing this consistently with the best practices listed below. You shouldn't be using HttpRuntime.Cache or HttpContext.Current.Cache directly, you should always be accessing it via the ApplicationContext.ApplicationCache object (`Umbraco.Core.CacheHelper`).
+You should always be doing this consistently with the best practices listed below. You shouldn't be using HttpRuntime.Cache or HttpContext.Current.Cache directly, you should always be accessing it via the ApplicationContext.ApplicationCache object (`Umbraco.Core.Cache.AppCaches`).
 
 ## Cache types
 
-The `Umbraco.Core.CacheHelper` which can be found in namespace `Umbraco.Core.Cache` contains 3 types of cache: Runtime Cache, Request Cache and Static Cache.
+The `AppCaches` which can be found in namespace `Umbraco.Core.Cache` contains servaral types of cache: Runtime Cache, Request Cache and Isolated Cache.
```
Comment (sofietoft) [inline Reference/Cache/applicationcache.md:11]:
```suggestion
The `AppCaches` which can be found in namespace `Umbraco.Core.Cache` contains several types of cache: Runtime Cache, Request Cache and Isolated Cache.
```

File: Reference/Cache/applicationcache.md [hunk 3]
```diff
@@ -1,31 +1,57 @@
...
 # Accessing the cache
 
-:::warning
-Please be aware that **this article has not yet been verified and updated against Umbraco 8**.
-
-The documentation available around caching in the current Umbraco version can be found here: [Caching](../Cache).
-:::
-
-You should always be doing this consistently with the best practices listed below. You shouldn't be using HttpRuntime.Cache or HttpContext.Current.Cache directly, you should always be accessing it via the ApplicationContext.ApplicationCache object (`Umbraco.Core.CacheHelper`).
+You should always be doing this consistently with the best practices listed below. You shouldn't be using HttpRuntime.Cache or HttpContext.Current.Cache directly, you should always be accessing it via the ApplicationContext.ApplicationCache object (`Umbraco.Core.Cache.AppCaches`).
```
Comment (sofietoft) [inline Reference/Cache/applicationcache.md:7]:
```suggestion
You should always be doing this consistently with the best practices listed below. You shouldn't be using HttpRuntime.Cache or HttpContext.Current.Cache directly. Instead, you should always be accessing it via the ApplicationContext.ApplicationCache object (`Umbraco.Core.Cache.AppCaches`).
```

**Review by sofietoft — COMMENTED** [review]
Thanks for updating this article @erikjanwestendorp !
Much appreciated.

I've made some suggestion - some sentences were a bit long, so I've tried to shorten them. Let me know if something doesn't look right.

Also, I'll make sure this is reviewed by our developers as soon as possible 💪

**Comment by erikjanwestendorp** [comment]
@sofietoft Thanks for your review. Just updated the PR!

**Comment by sofietoft** [comment]
Great. Thanks @erikjanwestendorp !

I've also sent it to our developers, and they've set off some time next week, to give this a quick review.

File: Reference/Cache/applicationcache.md [hunk 4]
```diff
@@ -1,31 +1,57 @@
...
+
+    private IAppPolicyCache _runtimeCache;
+    private IAppCache _requestCache;
+    private IsolatedCaches _isolatedCaches;
+	
+    public MyClass(AppCaches appCaches)
+    {
+        _runtimeCache = appCaches;
+        _requestCache = appCaches;
+        _isolatedCaches = appCaches;
```
Comment (elit0451) [inline Reference/Cache/applicationcache.md:42]:
```suggestion
        _runtimeCache = appCaches.RuntimeCache;
        _requestCache = appCaches.RequestCache;
        _isolatedCaches = appCaches.IsolatedCaches;
```

**Review by elit0451 — COMMENTED** [review]
Hey @erikjanwestendorp 🙂 

Thank you very much for updating this article 🎉
I've made a few improvement comments otherwise it looks good 💪

File: Reference/Cache/applicationcache.md (hunk 4 above)
Comment (erikjanwestendorp) [reply Reference/Cache/applicationcache.md:42]: Oops.. 🙈

**Comment by erikjanwestendorp** [comment]
@elit0451 Thanks for your review 😄! @sofietoft I updated the PR based on the given suggestions.

**Comment by sofietoft** [comment]
That's great @erikjanwestendorp !

Looks like we have the green light to merge this in then.
Thanks again for updating the article 💪

**Pull request merged**